pkg/reconciler/reconcile.go

```diff
@@ -9,7 +9,22 @@ import (
 	"go.uber.org/zap"
 )
 
-func Reconcile(logger *zap.Logger, client *db.Database, req forman.Request) forman.Result {
+type FleetReconciler struct {
+	// Data required for reconcile phases
+}
+
+func (r *FleetReconciler) Reconcile(ctx context.Context) (forman.Result, error) {
+	// Reconciles the desired state
+	// Example for an installation flow:
+	// 	1. Ensures the infra via AWS Cloudformation
+	// 	2. Ensures ROKS Cluster installation on existing infra
+	// 	3. Ensures ManagedFusionAgent Operator on ROKS Cluster
+	// 	4. Ensures presence of config for MFAO monitoring stack
+	// 	5. Ensures Data Foundation Config is present on ROKS Cluster
+	return forman.Result{}, nil
+}
+
+func Reconcile(logger *zap.Logger, r *FleetReconciler, client *db.Database, req forman.Request) forman.Result {
```

Comment:
Getting the reconciler from the outside does not make sense, as it should be a local instance to this specific reconcile to hold this specific reconcile context

Reply:
done.